synapse/handlers/room_member.py

```diff
@@ -235,9 +289,25 @@ async def _local_membership_update(
 
         if event.membership == Membership.JOIN:
             newly_joined = True
+            is_invite = False
```

Comment:
`is_invite` implies that the current event is an invite. Suggest `user_is_invited` or somesuch